tests/message_streaming_decoder_test.c

```diff
@@ -51,10 +51,14 @@ static void s_decoder_test_on_prelude_received(
         aws_mem_release(decoder_data->alloc, decoder_data->latest_payload);
     }
 
-    decoder_data->latest_payload = aws_mem_acquire(
-        decoder_data->alloc,
-        decoder_data->latest_prelude.total_len - AWS_EVENT_STREAM_PRELUDE_LENGTH - AWS_EVENT_STREAM_TRAILER_LENGTH -
-            decoder_data->latest_prelude.headers_len);
+    const size_t payload_size = decoder_data->latest_prelude.total_len - AWS_EVENT_STREAM_PRELUDE_LENGTH -
+                                AWS_EVENT_STREAM_TRAILER_LENGTH - decoder_data->latest_prelude.headers_len;
+
+    if (payload_size) {
```

Comment:
Would it be better to write this as ```decoder_data->latest_payload = payload_size ? aws_mem_acquire(decoder_data->alloc, payload_size) : NULL)```

Reply:
I don't think it makes a huge difference, but I do think it's just a bit clearer to separate the cases when one of them performs something expensive like malloc.

Reply:
Because of the side effects, I think the current branching is a bit more clear. My $0.02.